### DISCUSSION TITLE
GitHub files for upload

### DISCUSSION THREAD

**Discussion by MelanX** [Kira-NT/mc-publish#122] [Q&A] [answered]
When triggering a workflow that uses this action, the version, changelog, and name are extracted from the GitHub release.
Is there also a way to select one of the files of this release to be published to Modrinth/CurseForge?

**Comment by Kira-NT** [comment]
There's no built-in way to do that, as users usually upload the same local files both to GitHub and other platforms.

While I strongly recommend setting up a fully-featured CI that builds distributable files from scratch, instead of building those on your machine and then propagating them to platforms like GitHub, Modrinth, and/or CurseForge, so we can better avoid Fractureiser-like situations in the future, I understand that everybody's workflows differ. So, if you really need to propagate files manually uploaded to a GitHub Release, you can always use something like [robinraju/release-downloader](https://github.com/robinraju/release-downloader) to download release files and then pass those back to mc-publish!

**Reply by MelanX** [reply]
Thanks for the answer! I wasn’t fully sure how exactly everything would work correctly with setting versions and stuff like that. My current workflow is that I create a GitHub release using one workflow. And the workflow for uploading to Modrinth triggers on that GitHub release. For that, I download the file using the release downloader.

**Reply by Kira-NT** [reply]
That's an interesting setup for sure! :D Just as a heads up, `mc-publish` is also capable of creating GitHub releases, so you might be able to simplify your workflow in that regard. Other than that, yeah, use `release-downloader` to first download the files and then pass them to `mc-publish` as usual.

**Reply by MelanX** [reply]
Ahh I need to clarify that this is for a modpack, and that’s why I’m struggling with setting the correct version. Maybe it was something else where I didn’t know how to set it correctly but there was definitely something why I needed it that way. Because your readme said something like "data pulled from gh release", and that’s why I made a release trigger workflow. Not sure what it was :D But thanks for your help! 

If you’re curious why I need to build a modpack like that: https://github.com/ChaoticTrials/CaveStone/tree/1.20.1

**Reply by Kira-NT** [reply, marked as answer]
> "data pulled from gh release"

It's not mandatory by any means! :)

I looked into your workflows and can confirm that you can safely merge them into one. Moreover, you can replace `ncipollo/release-action` with `mc-publish` and publish to both Modrinth and GitHub in one go.

**Reply by MelanX** [reply]
Awesome, thank you!

**Reply by Kira-NT** [reply]
You are welcome! Feel free to reach out if you have more questions :)

**Reply by MelanX** [reply]
Ahh there's one more thing! I want to upload just one file to Modrinth (`*.mrpack`), but two files to GitHub (`*.zip`). How would I do this?

**Reply by Kira-NT** [reply]
`files` is a shared property, but you can always use more specialized ones:

```yaml
modrinth-files: *.mrpack
github-files: |
  *.mrpack
  *.zip
```

**Reply by MelanX** [reply]
Oh, nice to know! Thanks!
